docs/overloading-wp-behavior.md

```diff
@@ -124,56 +124,114 @@ The `@snapwp/blocks` package provides a large and growing number of [Block Compo
+		backgroundColor,
+		color: textColor,
+		fontSize,
+	};
 
-function MyCustomParagraph( { attributes } ) {
 	return (
-		<div className="my-custom-paragraph">
-			<p>{ attributes.content }</p> { /* Access block attributes */ }
-			{ /* Add any additional JSX or logic here */ }
-			<span className="custom-attribution"> - My Custom Rendering</span>
-		</div>
+		<p className="my-custom-paragraph" style={ styleObject }>
+			{ content }
+		</p>
 	);
-}
+};
 
 export default MyCustomParagraph;
 ```
 
-### 3. Mapping the Block
+### 3. Applying the Override
 
-Map your custom component to the block in your SnapWP configuration.
+SnapWP provides two ways to apply custom block overrides:
 
-```tsx
+-   Global Override via `snapwp.config.ts`
+-   Per-Route Override by passing a prop to `EditorBlocksRenderer`
+
+**Global Override (snapwp.config.ts)**
+
+To apply your block override globally, update your SnapWP configuration:
+
+```ts
+// snapwp.config.ts
+import type { SnapWPConfig } from '@snapwp/core/config';
+import MyCustomParagraph from './src/app/core-paragraph';
+
+const config: SnapWPConfig = {
+	blockDefinitions: {
+		CoreParagraph: MyCustomParagraph,
+	},
+};
+
+export default config;
+```
+
+With this setup, `MyCustomParagraph` will be used across all pages where the Core Paragraph block is rendered.
+
+**Per-Route Override (Prop to EditorBlocksRenderer)**
+
+If you want to override blocks only for a specific page or route, pass a `blockDefinitions` prop to `EditorBlocksRenderer` instead:
+
+```ts
 import { TemplateRenderer } from '@snapwp/next';
 import MyCustomParagraph from './components/MyCustomParagraph';
 import { EditorBlocksRenderer } from '@snapwp/blocks';
 
-const blockDefinitions = {
-	CoreParagraph: MyCustomParagraph, // Overload CoreParagraph
-	CoreImage: null, // Use default rendering for CoreImage
+const pageBlockDefinitions = {
+	CoreParagraph: MyCustomParagraph, // Override only for this page
 };
 
 export default function Page() {
 	return (
 		<TemplateRenderer>
-			{ ( editorBlocks ) => (
+			{(editorBlocks) => (
 				<EditorBlocksRenderer
-					editorBlocks={ editorBlocks }
-					blockDefinitions={ blockDefinitions }
+					editorBlocks={editorBlocks}
+					blockDefinitions={pageBlockDefinitions}
 				/>
-			) }
+			)}
 		</TemplateRenderer>
 	);
 }
 ```
 
+This allows you to apply the override only on specific routes while using the default block rendering elsewhere.
+
 Now, whenever a `core/paragraph` block is encountered, your `MyCustomParagraph` component will be used to render it. Any other blocks will use the default rendering unless you provide a custom component in `blockDefinitions`.
 
+> [!TIP]
+> If a block is overridden both globally (`snapwp.config.ts`) and per-route (`EditorBlocksRenderer` prop), the per-route override takes precedence.
+
 > [!TIP]
 > Setting the block definition value to `null` will use the [default block rendering](#default-html-rendering-with-parse-and-html-react-parser).
 
```

Comment:
In context, this should now move up to the global overrides section, and include a `blockDefinitions: {...}` snippet.

Then, in the per-route example, you can just inline the `CoreSomethingElse: null, // Use html-react-parser instead of a Block Component.` into the existing snippet.